### PR TITLE
BREAKING CHANGE: upgrade to PowerFactory 2024

### LINKED ISSUE #150: BREAKING CHANGE: upgrade to PowerFactory 2024
PF 2023 introduces some really helpful fixes and additionally starts supporting Python 3.11

### PR DESCRIPTION
**Pull request by SebastianDD**
Fixes #150

### CODE REVIEW

**Comment by SebastianDD** [comment]
consider to build python-version-specific wheels or to add in readme `uv sync --python 3.12` as example code to specify the python version to be used

**Review by sasanjac — CHANGES_REQUESTED** [review]

File: src/powerfactory_tools/utils/io.py [hunk 1]
```diff
@@ -93,30 +90,12 @@ def export_user_data(
     def create_file_path(
         self,
         *,
-        file_type: FileType,
-        file_name: str | None = None,
-        active_study_case: PFTypes.StudyCase | None = None,
+        file_type: FileType,  # noqa: ARG002
+        file_name: str | None = None,  # noqa: ARG002
+        active_study_case: PFTypes.StudyCase | None = None,  # type: ignore  # noqa: F821, PGH003, ARG002
     ) -> pathlib.Path:
```
[the remark this answers is not in the document]
Comment (SebastianDD) [reply src/powerfactory_tools/utils/io.py:100]: done

File: src/powerfactory_tools/__version__.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # :author: Sasan Jacob Rasti <sasan_jacob.rasti@tu-dresden.de>
 # :author: Sebastian Krahmer <sebastian.krahmer@tu-dresden.de>
-# :copyright: Copyright (c) Institute of Electrical Power Systems and High Voltage Engineering - TU Dresden, 2022-2023.
+# :copyright: Copyright (c) Institute of Electrical Power Systems and High Voltage Engineering - TU Dresden, 2022-2024.
```
[the remark this answers is not in the document]
Comment (SebastianDD) [reply src/powerfactory_tools/__version__.py:3]: Fixes #267

File: src/powerfactory_tools/utils/io.py [hunk 3]
```diff
@@ -90,33 +90,16 @@ def export_user_data(
...
-            file_path.resolve()
-        except OSError as e:
-            msg = f"File path {file_path} is not a valid path."
-            raise FileNotFoundError(msg) from e
-
-        # Create (sub)direcotries if not existing
-        file_path.parent.mkdir(parents=True, exist_ok=True)
-
-        return file_path
+        msg = "This method should be overridden in a subclass"
```
Comment (sasanjac) [inline src/powerfactory_tools/utils/io.py:101]:
```suggestion
        msg = "This method should be implemented in a subclass"
```

File: src/powerfactory_tools/utils/io.py [hunk 4]
```diff
@@ -50,12 +48,14 @@ def has(cls, value: str) -> bool:
 
 
 @pydantic.dataclasses.dataclass
-class ExportHandler:
+class BaseExportHandler(abc.ABC):
```
Comment (sasanjac) [inline src/powerfactory_tools/utils/io.py:51]:
```suggestion
class ExportHandler(abc.ABC):
```

File: src/powerfactory_tools/versions/pf2022/utils/io.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+# :license: BSD 3-Clause
+
+from __future__ import annotations
+
+import datetime as dt
+import typing as t
+
+import pydantic
+
+from powerfactory_tools.utils.io import BaseExportHandler
```
Comment (sasanjac) [inline src/powerfactory_tools/versions/pf2022/utils/io.py:13]:
```suggestion
from powerfactory_tools.utils.io import ExportHandler as BaseExportHandler
```

File: src/powerfactory_tools/versions/pf2024/utils/io.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+# :license: BSD 3-Clause
+
+from __future__ import annotations
+
+import datetime as dt
+import typing as t
+
+import pydantic
+
+from powerfactory_tools.utils.io import BaseExportHandler
```
Comment (sasanjac) [inline src/powerfactory_tools/versions/pf2024/utils/io.py:13]:
```suggestion
from powerfactory_tools.utils.io import ExportHandler as BaseExportHandler
```

**Review by sasanjac — CHANGES_REQUESTED** [review]

**Review by sasanjac — APPROVED** [review]

**Pull request merged**